marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/artifacts/setArtifact.sjs

```diff
@@ -16,7 +16,18 @@
 'use strict';
 
 const Artifacts = require('/data-hub/5/artifacts/core.sjs');
+const ds = require("/data-hub/5/data-services/ds-utils.sjs");
 
 var artifactType, artifactName, artifact;
 
+if ("ingestion" === artifactType) {
+  xdmp.securityAssert("http://marklogic.com/data-hub/privileges/read-ingestion", "execute");
+} else if ("mapping" === artifactType) {
+  xdmp.securityAssert("http://marklogic.com/data-hub/privileges/read-mapping", "execute");
+} else if ("flow" === artifactType || "stepDefinition" === artifactType) {
+  xdmp.securityAssert("http://marklogic.com/data-hub/privileges/read-flow", "execute");
```

Comment:
Shouldn't all of this be 'write-*' instead of 'read-*'  as setArtifact() is a write operation ?

Reply:
Oops, that is correct, will fix.